pkg/proxy/iptables/proxier_test.go

```diff
@@ -1111,6 +1113,258 @@ COMMIT
 	}
 }
 
+// cidrContains checks if cidrStr contains ipStr. cidrStr can actually be a CIDR or an IP
+func cidrContains(cidrStr, ipStr string) bool {
+	ip := netutils.ParseIPSloppy(ipStr)
+	if ip == nil {
+		panic("bad IP in test case: " + ipStr)
+	}
+
+	if strings.Contains(cidrStr, "/") {
+		_, cidr, err := netutils.ParseCIDRSloppy(cidrStr)
+		if err != nil {
+			panic("bad CIDR in kube-proxy output: " + err.Error())
+		}
+		return cidr.Contains(ip)
+	} else {
+		ip2 := netutils.ParseIPSloppy(cidrStr)
+		if ip2 == nil {
+			panic("bad IP in kube-proxy output: " + cidrStr)
+		}
+		return ip.Equal(ip2)
+	}
+}
+
+var commentRegex = regexp.MustCompile(`-m comment --comment ("[^"]*"|[^" ]*) `)
+var moduleRegex = regexp.MustCompile("-m ([^ ]*)")
+var localRegex = regexp.MustCompile("(!)? --(src|dst)-type LOCAL")
+var destIPRegex = regexp.MustCompile("(!)? -d ([^ ]*) ")
+var destPortRegex = regexp.MustCompile(" --dport ([^ ]*) ")
+var sourceIPRegex = regexp.MustCompile("(!)? -s ([^ ]*) ")
+var affinityRegex = regexp.MustCompile(" --rcheck ")
+var probabilityRegex = regexp.MustCompile(" --probability ([^ ]*) ")
+var jumpToEndpointRegex = regexp.MustCompile(" -j KUBE-SEP-")
+
+var noMatchModules = sets.NewString("conntrack", "mark")
+var knownModules = sets.NewString("addrtype", "recent", "statistic", "tcp")
```

Comment:
he, we don't have any rule using `udp` or `sctp`

Reply:
that's what I was saying about this not being fully generic